### PR TITLE
Fix unaligned accesses and tweak some stuff for non-standard systems

### PR DESCRIPTION
**Pull request by giulianobelinassi**
Systems running old ARM systems (ARM7, ARM9) has issues when accessing
unaligned addresses, resulting in swapped values. This commit fixes most
(if not all) of the issues regarding Tiberian Dawn.

There is still unaligned access warnings on BigShpBuffer, but that is not
an issue because althrough it reads/writes wrong value in memory, this is
correctly swapped in the CPU.

This commit also fixes issues with obsucure filesystem implementations,
like fseek being slow when it should return imediatelly and not finding
upper-case files on FAT32.

This also fixes a leak on Tiberian Dawn scorescreen.

Signed-off-by: Giuliano Belinassi <giuliano.belinassi@gmail.com>

### CODE REVIEW

File: common/crc.cpp [hunk 1]
```diff
@@ -108,7 +109,9 @@ int32_t CRCEngine::operator()(void const* buffer, size_t length)
         int32_t const* longptr = (int32_t const*)dataptr;
         int longcount = bytes_left / sizeof(int32_t); // Whole 'long' elements remaining.
         while (longcount--) {
-            CRC = lrotl(CRC, 1) + *longptr++;
+            int32_t l;
+            memcpy(&l, longptr++, sizeof(int32_t));
```
Comment (OmniBlade) [inline common/crc.cpp:113]: If you use the `cstring` include, you should use `std::memcpy` as not all implementations guarantee that header will expose it in the global namespace. Alternatively use `string.h` header which does.

File: tiberiandawn/init.cpp [hunk 2]
```diff
@@ -348,7 +348,7 @@ bool Init_Game(int, char*[])
         }
 
         strcpy(scan_path, search_path);
-        strcat(scan_path, "SC*.MIX");
+        strcat(scan_path, "sc*.mix");
```
Comment (OmniBlade) [inline tiberiandawn/init.cpp:351]: How come the case is changed here?

File: tiberiandawn/loaddlg.cpp [hunk 3]
```diff
@@ -451,7 +451,7 @@ int LoadOptionsClass::Process(void)
             game_idx = listbtn.Current_Index();
             game_num = Files[game_idx]->Num;
             if (WWMessageBox().Process(TXT_DELETE_FILE_QUERY, TXT_YES, TXT_NO) == 0) {
-                sprintf(fname, "SAVEGAME.%03d", game_num);
+                sprintf(fname, "savegame.%03d", game_num);
```
Comment (OmniBlade) [inline tiberiandawn/loaddlg.cpp:454]: How come the case is changed here?

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: tiberiandawn/init.cpp (hunk 2 above)
Comment (giulianobelinassi) [reply tiberiandawn/init.cpp:351]: For some reason, some implementations (the Nintendo DS, for instance) isn't happy about uppercase wildcards when finding files, even if the filesystem should be case-insensitive (FAT32).

I can remove this from the PR if you'd prefer.

File: tiberiandawn/loaddlg.cpp (hunk 3 above)
Comment (giulianobelinassi) [reply tiberiandawn/loaddlg.cpp:454]: See above comment.

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**